src/app/modal/modal.controller.js

```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
+                    this.linkedInData.resume += education[i].degree + ' ';
+                }
+                // Add Field of Study Type
+                if (education[i].fieldOfStudy) {
+                    this.linkedInData.resume += education[i].fieldOfStudy + ' ';
+                }
+                // Add line break
+                if (education[i].degree || education[i].fieldOfStudy) {
+                    this.linkedInData.resume += lineBreak;
+                }
+                // Add School
+                if (education[i].schoolName) {
+                    this.linkedInData.resume += education[i].schoolName + ' ';
+                }
+                // Add Start Date
+                if (education[i].startDate) {
+                    this.linkedInData.resume += education[i].startDate.year + ' - ';
+                }
+                // Add End Date
+                if (education[i].endDate) {
+                    this.linkedInData.resume += education[i].endDate.year + ' ';
+                }
+                // Add line break
+                if ((education[i].schoolName || education[i].startDate || education[i].endDate) && education[i + 1]) {
+                    this.linkedInData.resume += lineBreak;
+                }
+            }
+            // Add line break
+            this.linkedInData.resume += hardBreak;
+        }
+
         // Work Experience Block
-        this.linkedInData.resume = this.$filter('i18n')('modal.workExperience') + lineBreak;
+        this.linkedInData.resume += this.$filter('i18n')('modal.workExperience') + lineBreak;
         // Positions
         if (userProfile.positions) {
-            if (userProfile.positions.values && userProfile.positions.values.length) {
-                this.linkedInData.resume += (userProfile.positions.values[0].company.name || '') + ' ';
-                // Start Date
-                if (userProfile.positions.values[0].startDate) {
-                    this.linkedInData.resume += months[userProfile.positions.values[0].startDate.month - 1] + ' ' + userProfile.positions.values[0].startDate.year + ' - ' || '';
-                }
-                // End Date or 'Present'
-                if (userProfile.positions.values[0].endDate) {
-                    this.linkedInData.resume += months[userProfile.positions.values[0].endDate.month - 1] + ' ' + userProfile.positions.values[0].endDate.year || '';
-                } else {
-                    if (userProfile.positions.values[0].isCurrent) {
-                        this.linkedInData.resume += this.$filter('i18n')('modal.present');
+            var positions = userProfile.positions.values;
+            // Iterate through each position
+            if (positions && positions.length) {
+                for (var ii = 0; ii < positions.length; ii++) {
+                    // Add Employee section header
+                    this.linkedInData.resume += (positions[ii].company.name || '') + ' ';
```

Comment:
Is the company field on the positions anonymous object guaranteed to by defined and non-null? Consider a defensive de-referencing expression:

`((positions[i].company || {}).name || ' ') + ' '`